### PR TITLE
Fixes #23772: Make rudder-package able to install plugins from the repository

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23772

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/rudder-package/src/lib.rs [hunk 1]
```diff
@@ -81,19 +83,47 @@ pub fn run() -> Result<()> {
...
+        PACKAGES_DATABASE_PATH, REPOSITORY_INDEX_PATH, RUDDER_VERSION_PATH, TMP_PLUGINS_FOLDER,
+    };
     use std::path::Path;
 
-    pub fn install(force: bool, package: String) -> Result<()> {
+    pub fn install(force: bool, package: String, repository: Repository) -> Result<()> {
         let rpkg_path = if Path::new(&package).exists() {
             package
         } else {
-            bail!("TODO");
```
Comment (amousset) [inline relay/sources/rudder-package/src/lib.rs:96]: just for information, there is a built-in `todo!()` macro for this purpose.

File: relay/sources/rudder-package/src/repo_index.rs [hunk 2]
```diff
@@ -1,24 +1,62 @@
...
+        Ok(index)
+    }
+
+    pub fn get_compatible_plugins(&self, webapp_version: RudderVersion) -> Vec<Plugin> {
+        self.clone()
+            .0
+            .into_iter()
+            .filter(|x| {
+                let distant_package_webapp_version =
+                    format!("{}", x.metadata.version.rudder_version);
```
Comment (amousset) [inline relay/sources/rudder-package/src/repo_index.rs:26]:
```suggestion
                    x.metadata.version.rudder_version.to_string();
```

is equivalent

**Review by amousset — APPROVED** [review]

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/rudder-package/src/lib.rs (hunk 1 above)
Comment (Fdall) [reply relay/sources/rudder-package/src/lib.rs:96]: :+1:

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**